extensions/builder.md

```diff
@@ -0,0 +1,145 @@
+# Builder
+
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
+
+## Table of Contents
+
+## Required File/Directories
+A builder MUST have the following directories/files:
+- `/cnb/order.toml` &rarr; As defined in the [platform specification][order-toml-spec]
+- `/cnb/stack.toml` &rarr; As defined in the [platform specification][stack-toml-spec]
```

Comment:
We should think about what happens when a builder contains a lifecycle that can support multiple platform APIs but we make a non-additive change to the schemas of these files between platform API versions.

Maybe we want to do something like `/cnb/<platform-api>/order.toml` and the lifecycle can choose the default based on the current API?

Reply:
Hmm. Would a change like that need an RFC? At the very least, to broadcast to builder creators. 

I personally don't think we would need to preface the directory with `<platform-api>` at this point, but I'm happy to hear push back on that.

Reply:
Is it worth bringing this up during a WG?

Reply:
It's definitely a change to the builder schema, so it could be helpful. I'd be interested in hearing from builder authors whether they think they will be creating multi-platform api builders or not.

Reply:
@jabrown85 thoughts?

Reply:
I don't think we are actively creating builders for multiple platform versions - it does happen to work with multiple versions. I don't dislike `/cnb/<platform-api>/order.toml` - but maybe suggest `/cnb/order.<platform-api>.toml` if we think we think we need this. Falling back to `/cnb/order.toml` if there are no more specific matches.

This problem is maybe a bit bigger than this RFC though. There are several defaults that reference local filesystem locations that could change between API versions, like defaulting to`/cnb/buildpacks`. If we changed that folder structure in vNext we would would run into similar trouble I think. The folder structure may only match one expected API version. The lifecycle would have to understand old and new structures regardless of the current platform API.

Reply:
Issue created: https://github.com/buildpacks/spec/issues/220